_posts/2018-12-13-arms-final-report.md

```diff
@@ -0,0 +1,151 @@
+
+# Syntax of ARMS
+
+The first and most important decision that was made about the design of ARMS was how we would like to write and parse it. It was initially decided that we were going to write our own syntax and parser so we could have maximum control over the syntax. Although, it was quickly found that writing a syntax, parser, and creating functionality all in one semester was not realistic.
+
+It was decided that having a working proof-of-concept was the highest priority to pursue first, so we decided to not write an original language and use an existing one that is well defined and supported. After some research, it was decided that TOML was the language that would support our needs and be the fasted to get started with.
+
+The key functionality that made TOML attractive was the concept of an 'Array of Tables' which provides us with syntax like this:
+
+```
+[[sphere]]
+name = "wheel1"
+radius = 0.5
+relative_position = [1.0, 1.0, 0.0]
+
+[[sphere]]
+name = "wheel2"
+radius = 0.5
+relative_position = [1.0, -1.0, 0.0]
+```
+
+This code illustrates how TOML creates arrays of dictionaries. These two '[[sphere]]' objects will be stored in one array as dictionaries with the values below them being key-value pairs. This allows Armature to organize all objects of the same type in a convenient manner and handle all of the same type of object simultaneously.
+
+TOML has proven not to be perfect for our uses though. There are some use cases where we wish to write ARMS in a certain way that is simply not supported by TOML. For example, while specifying joints in ARMS, you must include a 'parent' and 'child' attribute to specify which shapes that joint connects. Because the code is written sequentially this results in situations where a joint between two objects could be many lines away from the shape it is actually in relation too. Therefore, we contemplated that idea of using indentation to show the relationship between a parent and child shape. i.e. A child shape would be made the child of another shape by being indented one tab underneath the parent shape. But, the benefits of creating our own syntax and parser would have to be weighed against the cost of doing so. I would estimate that one entire semester of work would have to be done by someone who has taken the Language and Machines course at MSU to catch that version up to where ARMS is now.
+
+One great feature of the Creators is that they rely on having objects defined in a dictionary, but doesn't know of care about how that dictionary is made. So this potential future parser is completely decoupled from how the Creators create their files. The only limitation is the parser would have to create those dictionary objects in a way that the Creators understand.
```

Comment:
First mention of Creators. You should probably introduce the concept first.

Reply:
Done